builtin/logical/kv/path_metadata_test.go

```diff
@@ -58,10 +59,17 @@ func TestVersionedKV_Metadata_Put(t *testing.T) {
 	if resp.Data["cas_required"] != true {
 		t.Fatalf("Bad response: %#v", resp)
 	}
+	if resp.Data["metadata_cas_required"] != true {
+		t.Fatalf("Bad response: %#v", resp)
+	}
 	if resp.Data["delete_version_after"] != d.String() {
 		t.Fatalf("Bad response: %#v", resp)
 	}
 
+	if resp.Data["current_metadata_version"] != uint64(0) {
+		t.Fatalf("Expected initial current_metadata_version to be 0, got: %v", resp.Data["current_metadata_version"])
```

Comment:
```suggestion
	if resp.Data["current_metadata_version"] != uint64(1) {
		t.Fatalf("Expected initial current_metadata_version to be 1, got: %v", resp.Data["current_metadata_version"])
```

This will have a lot of other implications elsewhere fwiw.